HongCoin.sol

```diff
@@ -362,6 +363,11 @@ contract TokenCreation is TokenCreationInterface, Token, GovernanceInterface {
                 return;
         }
 
+        if (isFundReleased) {
```

Comment:
At this put, the value of isFundReleased might have been changed by the createTokenProxy logic.  Specifically, tryLockFund is called and may update the state of the isFundReleased variable. 

You are correct that createTokenProxy will throw if it is called and isFundreleased is already set to true (which is what we want it to do)